src/components/course/course-header/CourseRunCards.jsx

```diff
@@ -21,18 +22,30 @@ const CourseRunCards = () => {
   const {
     userSubsidyApplicableToCourse,
     missingUserSubsidyReason,
+    applicableCatalogUuid,
   } = useUserSubsidyApplicableToCourse();
+  const {
+    data: {
+      catalogList,
+      restrictedRunsAllowed,
+    },
+  } = useEnterpriseCustomerContainsContent([courseKey]);
   const { data: courseMetadata } = useCourseMetadata();
-  const { data: { catalogList } } = useEnterpriseCustomerContainsContent([courseKey]);
   const { data: { enterpriseCourseEnrollments } } = useEnterpriseCourseEnrollments();
   const { data: userEntitlements } = useUserEntitlements();
+  const availableCourseRuns = courseMetadata.availableCourseRuns.filter(r => isRunUnrestricted({
+    restrictedRunsAllowed,
+    courseMetadata,
+    courseRunKey: r.key,
+    applicableCatalogUuid,
+  }));
```

Comment:
[curious] I am wondering if we can pass `courseMetadata` directly to parse out the `courseRunKey` directly in the helper function from `availableCourseRuns`. That way we wouldn't need to pass a specific `courseRunKey`. 

Also if course metadata could be transformed in the `useCourseMetadata` hook, that would be ideal.
The reason being that `courseMetadata` would need to parse restricted runs for every usage of the hook versus abstracting that logic direct direct in `useCourseMetadata`.

The complexity comes from determining the `applicableCatalogUuid` downstream. A polarizing solution would be to allow a optional argument into the `useCourseMetadata` hook to pass in `applicableCatalogUuid` and if that argument is passed, we can determine whether we display the unrestricted runs. If the argument is not passed, we can probably default on not showing the restricted runs.

Reply:
In today's round of updates, I updated useCourseMetadata() to perform an initial pass at filtering the runs based only on restrictedRunsAllowed, but agnostic of the applicable catalog. I'll be honest, it seemed like the Right Thing to do, but I don't actually think it would have any effect on the resulting behavior.

I am intrigued, however, by Hamzah's polarizing proposal to add an optional argument into the `useCourseMetadata()` hook to pass in `catalogUuid`, as that would theoretically help centralize the filtering logic. Why am I getting the sense that this is taboo?

Reply:
I think passing a `catalogUuid` to use as a filter within `useCourseMetadata` could work; my biggest concern is just coupling `useCourseMetadata` to any subsidy-related queries _within_ the hook itself. Passing in the applicable `catalogUuid` seems reasonable, since it's not composing `useUserSubsidyApplicableToCourse` within `useCourseMetadata` directly, which is how I may have (incorrectly) interpreted the above suggestion originally.